src/modules/Dashboard/features/Participant/Assignments/AssignmentsTab/AssignmentsTab.hooks.test.tsx

```diff
@@ -0,0 +1,916 @@
+    targetSubject,
+    respondentSubject,
+    handleRefetch: mockHandleRefetch,
+    dataTestId: testId,
+  });
+
+  return (
+    <>
+      {targetSubject && (
+        <Button
+          onClick={() => onClickNavigateToData(activityOrFlow, targetSubject.id)}
+          data-testid={`${testId}-view-data`}
+        >
+          View Data
+        </Button>
+      )}
+      <ActionsMenu
+        menuItems={getActionsMenu(activityOrFlow)}
+        data-testid={`${testId}-activity-actions`}
+      />
+
+      {modals}
+    </>
+  );
+};
+
+/* Tests
+=================================================== */
+
+describe('useAssignmentsTab hook', () => {
+  beforeEach(() => {
+    mockUseFeatureFlags.mockReturnValue({
+      featureFlags: {
+        enableParticipantMultiInformant: true,
+        enableActivityAssign: true,
+      },
+      resetLDContext: jest.fn(),
+    });
+
+    mockGetRequestResponses({
+      [GET_APPLET_URL]: successfulGetAppletMock,
+      [GET_APPLET_ACTIVITIES_URL]: successfulGetAppletActivitiesMock,
+      [GET_WORKSPACE_RESPONDENTS_URL]: ({ userId }) => {
+        const respondents = [
+          mockedOwnerParticipant,
+          mockedFullParticipant1,
+          mockedLimitedParticipant,
+        ];
+        const filteredRespondents = userId
+          ? respondents.filter((r) => r.id === userId)
+          : respondents;
+
+        return mockSuccessfulHttpResponse<ParticipantsData>({
+          result: filteredRespondents,
+          count: filteredRespondents.length,
+        });
+      },
+      [GET_WORKSPACE_MANAGERS_URL]: mockSuccessfulHttpResponse<ManagersData>({
+        result: [mockedOwnerManager],
+        count: 1,
+      }),
+    });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('Edit', () => {
+    describe('should show or hide edit ability depending on role', () => {
+      test.each`
+        canEdit  | role                 | description
+        ${true}  | ${Roles.Manager}     | ${'editing for Manager'}
+        ${true}  | ${Roles.SuperAdmin}  | ${'editing for SuperAdmin'}
+        ${true}  | ${Roles.Owner}       | ${'editing for Owner'}
+        ${false} | ${Roles.Coordinator} | ${'editing for Coordinator'}
+        ${true}  | ${Roles.Editor}      | ${'editing for Editor'}
+        ${false} | ${Roles.Respondent}  | ${'editing for Respondent'}
+        ${false} | ${Roles.Reviewer}    | ${'editing for Reviewer'}
+      `('$description', async ({ canEdit, role }) => {
+        renderWithProviders(
+          <UseAssignmentsHookTest
+            activityOrFlow={mockParticipantActivities.autoAssignActivity}
+            targetSubject={mockedOwnerSubject}
+          />,
+          {
+            ...renderOptions,
+            preloadedState: { ...renderOptions.preloadedState, ...getPreloadedState(role) },
+          },
+        );
+
+        const actionDots = screen.queryByTestId(`${testId}-activity-actions-dots`);
+        if (actionDots) fireEvent.click(actionDots);
```

Comment:
If for whatever reason the test doesn't find the action dots, then they never get clicked and the action menu won't show up. Then the `canEdit=false` scenario becomes void.

I also suggest using `userEvent.click` over `fireEvent.click` in new tests where possible, as the testing library docs suggest that

```suggestion
        const actionDots = screen.getByTestId(`${testId}-activity-actions-dots`);
        await userEvent.click(actionDots);
```

Reply:
> If for whatever reason the test doesn't find the action dots, then they never get clicked and the action menu won't show up. Then the `canEdit=false` scenario becomes void.

If the action dots aren't found, that means all menu items were hidden (which may be the case for certain roles). I guess it doesn't make a lot of sense to then go and check if the desired menu item is then not found, but effectively the end result is the same: the menu item also doesn't exist if the action dots are missing altogether, so the test passes. I'll massage the logic slightly to skip the menu item check if the action dots are missing.

> I also suggest using `userEvent.click` over `fireEvent.click` in new tests where possible, as the testing library docs suggest that

I was on the fence about that. At first I did switch them all over to `userEvent` and saw how it was causing tests to then fail, even after I thought I'd introduced the required `await`s. Might have just not done it quite correctly. I figured it wasn't harmful to stick to `fireEvent` where possible, using `userEvent` only as needed, for the sake of simplicity/performance, but perhaps the performance impact is negligible and I should just suck it up and figure out why it caused tests to start failing. I'll take another look.

Reply:
Okay, I took your advice! Used `userEvent` in all places. Clearly I wasn't using it properly everywhere, so no wonder tests weren't always passing. I like that it allowed me to get away from using `waitFor` in several places. Thanks for the nudge @sultanofcardio!

Also improved the logic with the action dots.